### PR TITLE
add suggestion getter to  ActiveDataProvider,add suggestion unitTest

### PR DESCRIPTION
**Pull request by Julian-B90**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes/no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  |

### CODE REVIEW

File: ActiveDataProvider.php [hunk 1]
```diff
@@ -76,6 +76,31 @@ public function getAggregation($name)
...
+    /**
+     * Returns results of the specified suggestion.
+     * @param string $name suggestion name.
+     * @return array suggestion results.
+     * @throws InvalidCallException if query results do not contain the requested suggestion.
+     */
+    public function getSuggestion($name)
+    {
+        $suggestions = $this->getSuggestions();
+        Yii::info($suggestions);
```
Comment (beowulfenator) [inline ActiveDataProvider.php:97]: Why?

File: ActiveDataProvider.php [hunk 2]
```diff
@@ -76,6 +76,31 @@ public function getAggregation($name)
...
+        if (!isset($suggestions[$name])) {
+            throw new InvalidCallException("Suggestions '{$name}' not found.");
```
Comment (beowulfenator) [inline ActiveDataProvider.php:99]: How about `Suggestion '{$name}' not found`?

File: ActiveDataProvider.php (hunk 1 above)
Comment (Julian-B90) [reply ActiveDataProvider.php]: `Yii::info;` was remove

File: ActiveDataProvider.php (hunk 2 above)
Comment (Julian-B90) [reply ActiveDataProvider.php]: Yes this is a better error message.

**Pull request merged**